[Benchmark] Cholesky decomposition benchmark

zzzDavid opened the pull request:
## Description
This PR implements Cholesky decomposition algorithm in Bril. 

Cholesky decomposition transforms a Hermitian, positive-definite matrixinto the product of a  lower triangular matrix and its conjugate transpose. It is useful for numerical methods such as Monte Carlo Simulation or solving systems of linear equations.

sampsyo reviewed (changes requested):
This is a very ambitious benchmark; nice work!

Can you please add a bullet to the list in the documentation describing the benchmark?

zzzDavid commented:
> This is a very ambitious benchmark; nice work!
> 
> Can you please add a bullet to the list in the documentation describing the benchmark?

Thanks, I added the documentation, also fixed a bug. Now the Cholesky decomposition benchmark gets the same result as `scipy.linalg.cholesky`

sampsyo reviewed (approved):
Looks great; thanks!!

Pat-Lafon commented on benchmarks/cholesky.bril:
```diff
@@ -0,0 +1,335 @@
+.sum.done:
+  idx: int = mul row size;
+  idx: int = add idx col;
+  loc: ptr<int> = ptradd dest idx;
```
I think you meant `ptr<float>`

zzzDavid replied:
Thanks, this issue is resolved by this PR: https://github.com/sampsyo/bril/pull/158, please review.

Pat-Lafon commented on benchmarks/cholesky.bril:
```diff
@@ -0,0 +1,335 @@
+  a_ij_ptr: ptr<float> = ptradd arr1 ij_index;
+  b_ij_ptr: ptr<float> = ptradd arr2 ij_index;
+  jj_ptr: ptr<float> = ptradd arr2 jj_index;
+  a_ij = load a_ij_ptr;
+  b_ij = load b_ij_ptr;
+  b_jj = load jj_ptr;
+  value = fdiv a_ij b_jj;
```
`:float` ?

zzzDavid replied:
This issue is also fixed in PR https://github.com/sampsyo/bril/pull/158
